zio-cache/shared/src/main/scala/zio/cache/Cache.scala

```diff
@@ -256,7 +268,8 @@ object Cache {
                   value = map.putIfAbsent(k, MapValue.Pending(new MapKey(k), promise))
                 }
                 val result = if (value eq null) {
-                  lookupValueOf(in, promise)
+                  val rollbackResultIfError = if (rollbackIfError) Right(None) else Left(())
```

Comment:
Not sure this really makes sense. If the value isn't in the cache at all how is this any different from a `get` which will put the value in the cache even if it is an error? I think one of the issues here is we are treating `refresh` inconsistently with `get`.

Reply:
I see your point. To be consistent with `get` we can say that we put the error into the cache then and remove that special case.

Reply:
Yeah I think our principle should be that the idea, as I understand it, is that if there is a valid "good" value in the cache we don't want to overwrite it with a "bad" value. One question that raises is what we should do when the original value is itself an error.

Reply:
I do wonder whether this is the best path to address this. Like do you want to cache error values at all?

Reply:
- Empty cache
  - Lookup Success -> replace with lookup exit
  - Lookup Error -> replace with lookup exit
- Existing cache value
  - Lookup Success -> replace with lookup exit
  - Lookup Error
    - if the old one was a success, we keep it
    - if the old one was an error, do we keep it or replace with the new error? Replacing will reset the error ttl, so I think keeping the old one would be better since the main goal is to refresh the success value?